src/components/Articles.jsx

```diff
@@ -1,35 +1,65 @@
 import { useState, useEffect } from "react";
-
-import Card from "react-bootstrap/Card";
 import { getArticles } from "../utils/api";
+import { useParams, useSearchParams } from "react-router-dom";
 
 function Articles() {
   const [articles, setArticles] = useState([]);
   const [loading, isLoading] = useState(true);
+  const { topic } = useParams();
 
+  //conditional rendering??
+  // pass query into getArticles
   useEffect(() => {
-    getArticles().then(({ articles }) => {
+    getArticles(topic).then(({ articles }) => {
       setArticles(articles);
       isLoading(false);
     });
-  }, []);
+  }, [topic]);
 
   if (loading) {
     return <p id="loading">Loading...</p>;
   }
   return (
+    //CSS Magic
     <ul id="list-container">
       {articles.map((article) => {
+        let id = "";
+        if (article.topic === "cooking") {
+          id = "article-topic-cooking";
+        } else if (article.topic === "football") {
+          id = "article-topic-football";
+        } else if (article.topic === "coding") {
+          id = "article-topic-coding";
+        }
```

Comment:
This is kind of unnecessary, cooking, football,  and coding should all be unique enough.